File: src/main/java/com/github/lkqm/paper/repository/ProjectRepository.java
package com.github.lkqm.paper.repository;

import com.github.lkqm.paper.model.Project;

import java.util.List;

/**
 * ProjectRepository
 *
 * @author Mario Luo
 * @date 2019.01.19 11:23
 */
public interface ProjectRepository {

    Project findOne(String id);

    int save(Project project);

    List<Project> findAll();

    int delete(String id);
}

File: src/main/java/com/github/lkqm/paper/repository/ProjectRepository.kt
package com.github.lkqm.paper.repository

import com.github.lkqm.paper.model.Project

/**
 * ProjectRepository
 *
 * @author Mario Luo
 * @date 2019.01.19 11:23
 */
interface ProjectRepository {
    fun findOne(id: String): Project?
    fun save(project: Project): Int
    fun findAll(): List<Project>
    fun delete(id: String?): Int
}